plugins/usbdmx/AsyncPluginImpl.cpp

```diff
@@ -196,16 +203,28 @@ void AsyncPluginImpl::WidgetRemoved(VellemanK8062 *widget) {
 void AsyncPluginImpl::DeviceEvent(HotplugAgent::EventType event,
                                   struct libusb_device *device) {
   if (event == HotplugAgent::DEVICE_ADDED) {
-    USBDeviceAdded(device);
+    SetupUSBDevice(device);
   } else {
     USBDeviceID device_id = m_usb_adaptor->GetDeviceId(device);
-    DeviceState *state = STLFindOrNull(m_device_map, device_id);
-    if (state && state->factory) {
-      OLA_INFO << "USB device " << device_id << " removed";
-      state->factory->DeviceRemoved(&m_widget_observer, state->usb_device);
-      state->factory = NULL;
+    DeviceState *state;
+    if (!STLLookupAndRemove(&m_device_map, device_id, &state) || !state) {
+      return;
     }
-    STLRemoveAndDelete(&m_device_map, device_id);
+
+    // At some point we'll need to notify the factory here. For instance in the
```

Comment:
There isn't an actionable item. It's more of a 'someone will need to add code here sometime in the future'